table_schema_cache.go

```diff
@@ -232,33 +226,44 @@ func CompositePKError(schema, table string) error {
 	return fmt.Errorf("%s.%s attempted to default to a Composite Primary Key for Pagination purposes whereas this is not supported. Kindly specify a Pagination Key for this table in the CascadingPaginationColumnConfig", schema, table)
 }
 
-func (ts *TableSchema) paginationKeyColumn(dbname, table string, cascadingPaginationColumnConfig *CascadingPaginationColumnConfig) (*schema.TableColumn, error) {
+func NonNumericPaginationKeyError(schema, table, paginationKey string) error {
+	return fmt.Errorf("Pagination Key `%s` for %s.%s is non-numeric", paginationKey, schema, table)
+}
+
+func (t *TableSchema) paginationKeyColumn(dbname, table string, cascadingPaginationColumnConfig *CascadingPaginationColumnConfig, logger *logrus.Entry) (*schema.TableColumn, error) {
 	var err error
 	var paginationKeyColumn *schema.TableColumn
 
 	if paginationColumn, found := cascadingPaginationColumnConfig.PaginationColumnFor(dbname, table); found {
-		paginationKeyColumn = ts.findColumnByName(paginationColumn)
+		// Check that supplied pagination key column is in actual effect an existing column
+		paginationKeyColumn = t.findColumnByName(paginationColumn)
 		err = NonExistingPaginationKeyColumnError(dbname, table, paginationColumn)
-	} else {
-		if cascadingPaginationColumnConfig == nil || cascadingPaginationColumnConfig.FallbackColumn == "" {
-			switch len(ts.PKColumns) {
-			case 0:
-				err = NonExistingPKError(dbname, table)
-			case 1:
-				paginationKeyColumn = &ts.Columns[ts.PKColumns[0]]
-			default:
-				err = CompositePKError(dbname, table)
-			}
-		} else {
-			paginationKeyColumn = ts.findColumnByName(cascadingPaginationColumnConfig.FallbackColumn)
-			err = NonExistingPaginationKeyColumnError(dbname, table, cascadingPaginationColumnConfig.FallbackColumn)
+	} else if cascadingPaginationColumnConfig == nil || cascadingPaginationColumnConfig.FallbackColumn == "" {
+		// default to Primary Key if possible
+		switch len(t.PKColumns) {
+		case 0:
+			err = NonExistingPKError(dbname, table)
+		case 1:
+			paginationKeyColumn = &t.Columns[t.PKColumns[0]]
+		default:
+			err = CompositePKError(dbname, table)
 		}
+	} else {
+		// Check that supplied pagination key column to fallback to is in actual effect an existing column
+		paginationKeyColumn = t.findColumnByName(cascadingPaginationColumnConfig.FallbackColumn)
+		err = NonExistingPaginationKeyColumnError(dbname, table, cascadingPaginationColumnConfig.FallbackColumn)
 	}
 
 	if paginationKeyColumn == nil {
 		return nil, err
 	}
 
+	if paginationKeyColumn.Type != schema.TYPE_NUMBER {
+		err = NonNumericPaginationKeyError(dbname, table, paginationKeyColumn.Name)
+		logger.WithError(err).Error("invalid table")
```

Comment:
I don't think you actually need to log in this here. Probably should log the overall error at line 195, otherwise I don't think any of the other errors will be logged?

Reply:
Good catch!